src/asm/rgbasm.5

```diff
@@ -0,0 +1,426 @@
+obligation to follow any specific rules.
+The following example defines a section that can be placed anywhere in any ROMX
+bank:
+.Pp
+.Dl SECTION \*qCoolStuff\*q,ROMX
+.Pp
+If it is needed, the following syntax can be used to fix the base address of the
+section:
+.Pp
+.Dl SECTION \*qCoolStuff\*q,ROMX[$4567]
+.Pp
+It won't, however, fix the bank number, which is left to the linker.
+If you also want to specify the bank you can do:
+.Pp
+.Dl SECTION \*qCoolStuff\*q,ROMX[$4567],BANK[3]
+.Pp
+And if you only want to force the section into a certain bank, and not it's
+position within the bank, that's also possible:
+.Pp
+.Dl SECTION \*qCoolStuff\*q,ROMX,BANK[7]
+.Pp
+In addition, you can specify byte alignment for a section.
+This ensures that the section starts at a memory address where the given number
+of least-significant bits are 0.
+This can be used along with
+.Ar BANK No , if desired.
+However, if an alignment is specified, the base address must be left unassigned.
+This can be useful when using DMA to copy data or when it is needed to align the
+start of an array to 256 bytes to optimize the code that accesses it.
+.Pp
+.Dl SECTION \*qOAM Data\*q,WRAM0,ALIGN[8] ; align to 256 bytes
+.Pp
+.Dl SECTION \*qVRAM Data\*q,ROMX,BANK[2],ALIGN[4] ; align to 16 bytes
+.Pp
+HINT: If you think this is a lot of typing for doing a simple
+.Ar ORG No type thing you can quite easily write an intelligent macro (called
+.Ar ORG No for example) that uses Ar \@ No for the section name and determines
+correct section type etc as arguments for
+.Ar SECTION .
+.Pp
+Sections can also be placed by using a linkerscript file.
+The format is described in
+.Xr rgblink 5 .
+They allow the user to place floating sections in the desired bank in the order
+specified in the script.
+This is useful if the sections can't be placed at an address manually because
+the size may change, but they have to be together.
+.Pp
+.Sh SYMBOLS
+.Pp
+.Ss Symbols
+RGBDS supports several types of symbols:
+.Pp
+.Bl -hang
+.It Sy Label
+Used to assign a memory location with a name
+.It Sy EQUate
+Give a constant a name.
+.It Sy SET
+Almost the same as EQUate, but you can change the value of a SET during
+assembling.
+.It Sy Structure Po Sy the RS group Pc
+Define a structure easily.
+.It Sy String equate Pq Sy EQUS
+Give a frequently used string a name.
+Can also be used as a mini-macro, like #define in C.
+.It Sy MACRO
+A block of code or pseudo instructions that you invoke like any other mnemonic.
+You can give them arguments too.
+.El
+.Pp
+A symbol cannot have the same name as a reserved keyword.
+.Bl -hang
+.It Sy Label
+.Pp
+One of the assembler's main tasks is to keep track of addresses for you so you
+don't have to remember obscure numbers but can make do with a meaningful name, a
+label.
+.Pp
+This can be done in a number of ways:
+.Pp
+.Dl GlobalLabel
+.Dl AnotherGlobal:
+.Dl .locallabel
+.Dl .yet_a_local:
+.Dl ThisWillBeExported:: ;note the two colons
```

Comment:
For big chunks of code I use a display block.

    .Bd -literal
    Blah blah blah
    .Ed